UnityProject/Assets/Scripts/Managers/SOs.cs

```diff
@@ -0,0 +1,90 @@
+	public class SOs : SingletonManager<SOs>
+	{
+		// Put all singletons here (and in the dictionary below) and assign them in editor.
+		public AlcoholicDrinksSOScript AlcoholicDrinksSOScript;
+		public BestSlotForTrait BestSlotForTrait;
+		public CommonCooldowns CommonCooldowns;
+		public CommonPrefabs CommonPrefabs;
+		public CommonSounds CommonSounds;
+		public CommonTraits CommonTraits;
+		public CraftingRecipeSingleton CraftingRecipeSingleton;
+		public DepartmentList DepartmentList;
+		public GAS2ReagentSingleton GAS2ReagentSingleton;
+		public GasesSingleton GasesSingleton;
+		public GasMixesSingleton GasMixesSingleton;
+		public ItemTypeToTraitMapping ItemTypeToTraitMapping;
+		public MachinePartsItemTraits MachinePartsItemTraits;
+		public MachinePartsPrefabs MachinePartsPrefabs;
+		public OccupationList OccupationList;
+		public PipeTileSingleton PipeTileSingleton;
+		public PlayerTextureDataSOs PlayerTextureDataSOs;
+		public PoolConfig PoolConfig;
+		public RaceSOSingleton RaceSOSingleton;
+		public SOAdminJobsList AdminJobsList;
+		public SpeechModManager SpeechModManager;
+		public SpriteCatalogue SpriteCatalogue;
+		public SurgeryProcedureBaseSingleton SurgeryProcedureBaseSingleton;
+		public UIActionSOSingleton UIActionSOSingleton;
+		public UplinkCategoryList UplinkCategoryList;
+		public UplinkPasswordList UplinkPasswordList;
+
+		private Dictionary<Type, ScriptableObject> typeSOMap;
+
+		public override void Awake()
+		{
+			base.Awake();
+			typeSOMap = new Dictionary<Type, ScriptableObject>()
+			{
+				{ typeof(AlcoholicDrinksSOScript), AlcoholicDrinksSOScript },
+				{ typeof(BestSlotForTrait), BestSlotForTrait },
+				{ typeof(CommonCooldowns), CommonCooldowns },
+				{ typeof(CommonPrefabs), CommonPrefabs },
+				{ typeof(CommonSounds), CommonSounds },
+				{ typeof(CommonTraits), CommonTraits },
+				{ typeof(CraftingRecipeSingleton), CraftingRecipeSingleton },
+				{ typeof(DepartmentList), DepartmentList },
+				{ typeof(GAS2ReagentSingleton), GAS2ReagentSingleton },
+				{ typeof(GasesSingleton), GasesSingleton },
+				{ typeof(GasMixesSingleton), GasMixesSingleton },
+				{ typeof(ItemTypeToTraitMapping), ItemTypeToTraitMapping },
+				{ typeof(MachinePartsItemTraits), MachinePartsItemTraits },
+				{ typeof(MachinePartsPrefabs), MachinePartsPrefabs },
+				{ typeof(OccupationList), OccupationList },
+				{ typeof(PipeTileSingleton), PipeTileSingleton },
+				{ typeof(PlayerTextureDataSOs), PlayerTextureDataSOs },
+				{ typeof(PoolConfig), PoolConfig },
+				{ typeof(RaceSOSingleton), RaceSOSingleton },
+				{ typeof(SOAdminJobsList), AdminJobsList },
+				{ typeof(SpeechModManager), SpeechModManager },
+				{ typeof(SpriteCatalogue), SpriteCatalogue },
+				{ typeof(SurgeryProcedureBaseSingleton), SurgeryProcedureBaseSingleton },
+				{ typeof(UIActionSOSingleton), UIActionSOSingleton },
+				{ typeof(UplinkCategoryList), UplinkCategoryList },
+				{ typeof(UplinkPasswordList), UplinkPasswordList },
+			};
```

Comment:
```suggestion
			typeSOMap = new Dictionary<Type, ScriptableObject>()
			
			BindingFlags bindingFlags = BindingFlags.Public;

			foreach (FieldInfo field in typeof(SOs).GetFields(bindingFlags))
			{
				typeSOMap.Add(field.GetType(), field.GetValue(this));
			}
```

Think this'll work, needs testing though

Reply:
I was trying to avoid reflection. Dunno why I felt compelled as so.

Reply:
> I was trying to avoid reflection. Dunno why I felt compelled as so.

Yeah, understandable. Not a big deal to keep it as it currently is, just thought i'd point it out.

Reply:
It does look nicer I suppose. But yeah, can just get the reference directly from the manager in the future, skipping all this. Would be nice to not have to update this manager when adding new SOs.